### ISSUE TITLE
 .logs directory will not be created auto?

### ISSUE THREAD

**Issue by alongL** [gabime/spdlog#1248]
If i run example/example.cpp , basic_example() fails for  failed open "logs/basic-log.txt".
It means that .logs isn't created auto by program.
We should created this directory by ourself.
I think this is not smart. 
Will auto create directory be added?

```
bool createRecursionDir(std::string path)
{
    if (path.length() == 0) return true;
    std::string sub;
    fixPath(path);

    std::string::size_type pos = path.find('/');
    while (pos != std::string::npos)
    {
        std::string cur = path.substr(0, pos-0);
        if (cur.length() > 0 && !isDirectory(cur))
        {
            bool ret = false;
#ifdef WIN32
            ret = CreateDirectoryA(cur.c_str(), NULL) ? true : false;
#else
            ret = (mkdir(cur.c_str(), S_IRWXU|S_IRWXG|S_IRWXO) == 0);
#endif
            if (!ret)
            {
                return false;
            }
        }
        pos = path.find('/', pos+1);
    }

    return true;
}
```

**Comment by berge2007** [comment]
https://en.cppreference.com/w/cpp/filesystem/create_directory

**Comment by gabime** [comment]
spdlog is c++11

**Comment by gabime** [comment]
Fixed in 4a620a2c5e09646cf79cfd9000517ff0d2bb9edd
